pandas/core/arrays/numeric.py

```diff
@@ -126,7 +127,7 @@ def _standardize_dtype(cls, dtype) -> NumericDtype:
             # https://github.com/numpy/numpy/pull/7476
             dtype = dtype.lower()
 
-        if not issubclass(type(dtype), cls):
+        if not isinstance(dtype, NumericDtype):
```

Comment:
I think it is safe to check against `NumericDtype` and not `cls`: if it were a more specific sub-class, `np.dtype` doesn't work with instances of `NumericDtype`.

Reply:
(mypy and pyright do not support `issubclass(type(some_object), some_class)` for type narrowing)